### ISSUE TITLE
operator-based analysis

### ISSUE THREAD

**Issue by ajgoza** [NUFgroup/IBPM.jl#9]
- [ ] Global stability
- [ ] resolvent
- [ ] decide on framework 
  - [ ] create a type for linear versus nonlinear (maybe stored within IBModel struct) and use multiple dispatch to linearize the nonlinear term about the base state when "linear". 
  - [ ] construct linear system as linear map and get evals via eigs, or back out from time stepping?
  - [ ] what about the base state? (if w built the linear system as a linear map would be straightforward to incorporate newton raphson)

**Comment by jcallaham** [comment]
Started working on this today.  Following the framework above, I think the plan will be to use multiple dispatch within `nonlinear` to construct the linearized solver, then wrap it in a linear map.  We'll also need to introduce a `LinearizedIBProblem <: AbstractIBProblem` that can store a base state (and ideally precompute the relevant arrays for the linearized advection term).

I haven't gotten as far as testing yet, just planning out how the functions will break down.

@ajgoza next time we talk we should take a look at the plan for the modified `nonlinear` function and make sure it's correct physically.

**Comment by ajgoza** [comment]
Yes, sounds good (to the plan and to the intention to discuss the nonlinear function in our meeting)!

**Comment by jcallaham** [comment]
Merged back into main.  *Almost* good to go (at least for direct stability analysis), but I want to troubleshoot the asymmetry in the nonlinear term before we go any farther with it (see #32)
